src/Middleware/HttpLogging/src/HttpLoggingMiddleware.cs

```diff
@@ -15,26 +16,22 @@ internal sealed class HttpLoggingMiddleware
 {
     private readonly RequestDelegate _next;
     private readonly ILogger _logger;
+    private readonly IHttpLoggingInterceptor[] _interceptors;
     private readonly IOptionsMonitor<HttpLoggingOptions> _options;
-    private const int DefaultRequestFieldsMinusHeaders = 7;
-    private const int DefaultResponseFieldsMinusHeaders = 2;
     private const string Redacted = "[Redacted]";
 
-    /// <summary>
-    /// Initializes <see cref="HttpLoggingMiddleware" />.
-    /// </summary>
-    /// <param name="next"></param>
-    /// <param name="options"></param>
-    /// <param name="logger"></param>
-    public HttpLoggingMiddleware(RequestDelegate next, IOptionsMonitor<HttpLoggingOptions> options, ILogger<HttpLoggingMiddleware> logger)
+    public HttpLoggingMiddleware(RequestDelegate next, IOptionsMonitor<HttpLoggingOptions> options, ILogger<HttpLoggingMiddleware> logger,
+        IEnumerable<IHttpLoggingInterceptor> interceptors)
```

Comment:
What is the API for registering new `IHttpLoggingInterceptor` instances?

I think this is a power user feature, and I'm ok with it not having a nice API, but it should be documented.